### DISCUSSION TITLE
Handling Win64 program

### DISCUSSION THREAD

**Discussion by ghotik** [leecher1337/ntvdmx64#184] [General]
Hi. 
I'm currently maintaining DxWnd.exe, a Win32 program that injects a Win32 dll (dxwnd.dll) into Win32 program (usually video-games) to perform a wide set of tricks.
I know that it would be possible to replicate the same idea on a Win64 environment, building a Win64 program that inject a Win64 dll etc. etc.
But this would require a big effort to rewrite the GUI program and would also make it harder to use the tool because any user should be aware of the nature of the target program to run the proper GUI (n.d.r. this is actually an excuse, the main reason is that I'm lazy and I'm not going to rewrite all this GUI code).
The current Win32 GUI could easily detect the nature of the target program, but I wonder if it could be possible for a Win32 program to inject a Win64 dll. 
Any suggestion?

**Comment by leecher1337** [comment]
Hi,
I wonder: Is there even use for Win64 programs? I thought the main focus is on old 16/32bit games, are there even 64bit games around that require DxWnd?
For injection of Win64 DLL from 32bit application, see https://github.com/leecher1337/ntvdmx64/blob/master/ntvdmpatch/src/ldntvdm/ldntvdm/wow64inj.c

**Reply by ghotik** [reply]
In effect, most of the DxWnd compatibility tricks would be useless (if not harmful) in a recent Win64 game, but I keep receiving requests. Reasons for a 64 bit DxWnd, IMMO, would be these:
1) running the game in window for the very few games developed by lazy programmers that didn't foresee this possible usage
2) time stretching, very useful for game cheating
3) texture dumping and hacking (though there should be better tools for this)

honestly I can't think of anything more.
